pkg/ioutils/bytespipe.go

```diff
@@ -47,22 +51,31 @@ func NewBytesPipe(buf []byte) *BytesPipe {
 // It can allocate new []byte slices in a process of writing.
 func (bp *BytesPipe) Write(p []byte) (int, error) {
 	bp.mu.Lock()
-	defer bp.mu.Unlock()
```

Comment:
Was this significant in the benchmarking? I know that `defer` can sometimes slow things down. But if it wasn't a measurable speedup, I would rather keep the `defer` here.